src/big/big_float.cr

```diff
@@ -13,6 +13,8 @@ struct BigFloat < Float
   end
 
   def initialize(str : String)
+    # Strip leading '+' char to smooth out cases with strings like "+123"
+    str = str.lchop('+')
```

Comment:
It would be really simple to optimize this by simply adding 1 to the pointer if `str.starts_with? '+'`. I think this optimization is so easy it might be a good idea to do it now to avoid the string copy and extra garbage.

Reply:
yeah, it's done in `String#lchop`.

Reply:
@Sija What do you mean? `lchop` creates a new string. We should just ignore the first char if it's `+`, like what @RX14 says.

Reply:
My bad indeed. `str = str[1..-1] if str.starts_with?('+')`?

Reply:
Wouldn't be better to implement it directly in `String#lchop` instead?

Reply:
It's not possible. All strings have 8 leading bytes: four bytes for the type_id, and four bytes for the string's size. So a substring must have this info too.

In any case, I consider this an optimization. We can merge this now and optimize it later.

Reply:
Yes, it's an optimization. However, we'll probably forget if this is merged as-is. This is a very simple optimization:
```cr
unsafe_str = str.to_unsafe
unsafe_str += 1 if str.starts_with?('+')
```

then use `unsafe_str`.

Reply:
I'd like to merge it as-is and leave this perf optimization to the most interested parties like yourself.

Reply:
I general I hate it that we need such unsafe code. We should maybe work with slices in most cases unless we can prove a big performance loss.